Support for packaging BNG with Anaconda

alubbock opened the pull request:
This PR introduces a set of changes needed to put BNG in an Anaconda package on Windows. The Anaconda build system uses either the Microsoft VS compiler (there's too much posix dependency in BNG, so this seems like a lot of work) or the [msys2](http://www.msys2.org/) environment, which is what I went with.

Itemized changes:

- Muparser is moved to the Network3/src/muparser directory, rather than compiled separately. This was necessary because the muparser build scripts wouldn't easily support msys2. The upside is that this speeds up the build time, as there's not a separate configure/make etc. for muparser.
- Replace the posix-specific CPU timing code with a cross-platform version in `cputime.cpp`, which works natively on Windows and posix using a bunch of `ifdef`s
- Update CVODE's `config.guess` and `config.sub`, which identify the build platform. They weren't updated since 2005, so they couldn't identify msys2. I updated them as described [here](https://www.gnu.org/software/gettext/manual/html_node/config_002eguess.html) and included them in a new `cvode-2.6.0.tar.gz`.
- Change `bng2/Network3/src/util/mathutils/random.c` to use the cross platform `rand()` and `srand()`, rather than the posix-specific `random()` and `srandom()`.
- Change `bng2\Perl2\BNGModel.pm` to search for batch files when looking for external executables on Windows (in addition to `.exe`s). Anaconda uses `.bat` files in place of symbolic links on Windows.

The result of all this is a simple unified build system. To build, clone my [Anaconda recipes](https://github.com/alubbock/anaconda-recipes/tree/master/bionetgen) repo, change to the `bionetgen` directory, and type `conda-build .`. Works on Windows, Linux, and Mac.

To try the BNG build using these changes (together with NFsim and PySB builds), see my [anaconda page](https://anaconda.org/alubbock/bionetgen) for the packages and installation instructions.

jrfaeder commented:
Hi Alex,

Sorry about the delay in addressing this. We resolved the build problems on Travis but there is a problem remaining on Appveyor with the Windows build that we would like you to address if possible before we do the merge.

Thanks,

Jim

alubbock commented:
Hi Jim,

I think the Appveyor build issues stem from the fact I was using Anaconda/msys2 rather than Cygwin for Windows builds (see e.g. [msys2 vs cygwin](https://github.com/msys2/msys2/wiki/How-does-MSYS2-differ-from-Cygwin) if you're not familiar with msys2).

I rebased my PR against BNG's latest master and updated appveyor.yml with this msys2 configuration. Unfortunately it's not building at the moment due to the private FTP you're using for fetching NFsim and Atomizer, which doesn't work with my fork. Hopefully it should work or be close to working on Appveyor now with access to that FTP server.

Let me know if anything else comes up.

Cheers,
Alex
